concordium-consensus/src/Concordium/GlobalState/Persistent/Accounts.hs

```diff
@@ -59,26 +69,28 @@ data Accounts (pv :: ProtocolVersion) = Accounts {
     -- |Unique index of accounts by 'AccountAddress'
     accountMap :: !(AccountMap.PersistentAccountMap pv),
     -- |Hashed Merkle-tree of the accounts
-    accountTable :: !(LFMBTree AccountIndex HashedBufferedRef (PersistentAccount (AccountVersionFor pv))),
+    accountTable :: !(LFMBTree' AccountIndex HashedBufferedRef (LazilyHashedCachedRef (AccountCache (AccountVersionFor pv)) (PersistentAccount (AccountVersionFor pv)))),
```

Comment:
I would like to have a description of the caching approach written in the comment to this data structure.

In particular the use of BlobRef's for keys in the cache is not that great since we maintain many copies of the same account, only one of which is relevant. In some scenarios this is not an issue, in others it reduces the effective size of the cache a lot.

These tradeoffs and the reason for the choices need to be documented so that we may understand them in the future.